# Data Understanding — accidentologie0.csv

Paris traffic accidents dataset (2019–2024). Each row = one person involved in one accident.

In [1]:
!pip install pandas


[notice] A new release of pip is available: 24.3.1 -> 26.1.2
[notice] To update, run: pip install --upgrade pip


In [2]:
import pandas as pd

In [3]:
df = pd.read_csv("../data/accidentologie0.csv", sep=";")

In [4]:
df.head(10)

,Date,IdUsager,PV,Arrondissement,Mode,Catégorie,Gravité,Age,Genre,Milieu,...,Nom arrondissement,Arrondissement.1,arrondgeo,Adresse 1,Adresse 2,Coordonnées.1,Nom arrondissement.1,Coordonnées.2,Nom arrondissement.2,Arrondissement.2
0,2024-10-16,9630806,7435,75120,2 Roues Motorisées,Conducteur,Blessé léger,28.0,Masculin,En-Agg,...,NaN,75120,"{""coordinates"": [[[[2.415286456, 48.855177828]...",AVENUE DU DOCTEUR GLEY,RUE DES FRERES FLAVIEN,"48.876801, 2.411995",Paris 20e Arrondissement,"48.876801, 2.411995",Paris 20e Arrondissement,75120.0
1,2024-10-19,9647261,7523,75111,2 Roues Motorisées,Conducteur,Blessé léger,24.0,Masculin,En-Agg,...,NaN,75111,"{""coordinates"": [[[[2.369123881, 48.853166231]...",Rue Mercoeur,Rue Léon Frot,"48.857625, 2.384619",Paris 11e Arrondissement,"48.857625, 2.384619",Paris 11e Arrondissement,75111.0
2,2024-10-26,9654580,7725,75110,EDP-m,Conducteur,Blessé léger,20.0,Masculin,En-Agg,...,NaN,75110,"{""coordinates"": [[[[2.363872117, 48.867516287]...",Rue de la Fidélité,BOULEVARD DE MAGENTA,"48.874804, 2.357233",Paris 10e Arrondissement,"48.874804, 2.357233",Paris 10e Arrondissement,75110.0
3,2024-11-26,9680888,8373,75119,Piéton,Piéton,Blessé léger,26.0,Masculin,En-Agg,...,NaN,75119,"{""coordinates"": [[[[2.410820319, 48.878436176]...",RUE DU MAROC,Rue d'Aubervilliers,"48.887327, 2.367062",Paris 19e Arrondissement,"48.887327, 2.367062",Paris 19e Arrondissement,75119.0
4,2024-11-29,9682542,8467,75114,Vélo,Conducteur,Blessé léger,21.0,Masculin,En-Agg,...,NaN,75114,"{""coordinates"": [[[[2.336656992, 48.839671108]...",BOULEVARD JOURDAN,NaN,"48.82174, 2.332572",Paris 14e Arrondissement,"48.82174, 2.332572",Paris 14e Arrondissement,75114.0
5,2024-12-19,9727496,9017,75116,4 Roues,Conducteur,Blessé léger,20.0,Masculin,En-Agg,...,NaN,75116,"{""coordinates"": [[[[2.245703788, 48.876459232]...",ALLEE DU BORD DE L EAU,NaN,"48.868477, 2.235562",Paris 16e Arrondissement,"48.868477, 2.235562",Paris 16e Arrondissement,75116.0
6,2020-05-28,202504824,3625,75120,2 Roues Motorisées,Conducteur,Blessé léger,50.0,Masculin,En-Agg,...,Paris 20e Arrondissement,75120,"{""coordinates"": [[[[2.415286456, 48.855177828]...",NaN,NaN,"48.857638, 2.403904",Paris 20e Arrondissement,"48.857638, 2.403904",Paris 20e Arrondissement,75120.0
7,2020-07-01,202511309,4849,75115,2 Roues Motorisées,Conducteur,Blessé léger,47.0,Feminin,En-Agg,...,Paris 15e Arrondissement,75115,"{""coordinates"": [[[[2.289407656, 48.828333842]...",NaN,NaN,"48.82788, 2.29139",Paris 15e Arrondissement,"48.82788, 2.29139",Paris 15e Arrondissement,75115.0
8,2020-07-03,202515742,4923,75115,4 Roues,Conducteur,Blessé léger,47.0,Feminin,En-Agg,...,Paris 15e Arrondissement,75115,"{""coordinates"": [[[[2.289407656, 48.828333842]...",NaN,NaN,"48.832118, 2.280614",Paris 15e Arrondissement,"48.832118, 2.280614",Paris 15e Arrondissement,75115.0
9,2019-08-31,181201190,8295,75111,2 Roues Motorisées,Conducteur,Blessé léger,41.0,Masculin,En-Agg,...,Paris 11e Arrondissement,75111,"{""coordinates"": [[[[2.369123881, 48.853166231]...",NaN,NaN,"48.856444, 2.381237",Paris 11e Arrondissement,"48.856444, 2.381237",Paris 11e Arrondissement,75111.0


**Observations:** Each row represents one person involved in an accident. Key fields include the transport mode (`Mode`), user category (`Catégorie`), injury severity (`Gravité`), and demographics (`Age`, `Genre`). The dataset spans multiple Paris arrondissements and contains both structured fields and free-text descriptions (`Résumé`). Some columns like `Nom arrondissement` have missing values for newer records (2024), while duplicates appear across coordinate and address columns — indicating possible join artifacts.

In [5]:
df.columns.tolist()

['Date',
 'IdUsager',
 'PV',
 'Arrondissement',
 'Mode',
 'Catégorie',
 'Gravité',
 'Age',
 'Genre',
 'Milieu',
 'Adresse',
 'Longitude',
 'Latitude',
 'Id accident',
 "Tranche d'age",
 'Blessés Légers',
 'Blessés hospitalisés',
 'Tué',
 'Résumé',
 'Coordonnées',
 'Nom arrondissement',
 'Arrondissement.1',
 'arrondgeo',
 'Adresse 1',
 'Adresse 2',
 'Coordonnées.1',
 'Nom arrondissement.1',
 'Coordonnées.2',
 'Nom arrondissement.2',
 'Arrondissement.2']

**Observations:** The dataset contains 30 columns with several duplicates from bad joins: `Nom arrondissement` / `.1` / `.2`, `Coordonnées` / `.1` / `.2`, `Arrondissement` / `.1` / `.2`, and `Adresse` / `Adresse 1` / `Adresse 2`. The core columns for analysis are: `Date`, `IdUsager`, `PV`, `Arrondissement`, `Mode`, `Catégorie`, `Gravité`, `Age`, `Genre`, `Milieu`, `Longitude`, `Latitude`, `Id accident`, `Tranche d'age`, and the three injury flags (`Blessés Légers`, `Blessés hospitalisés`, `Tué`).

## 1. Shape & dtypes

In [6]:
print(f"Rows: {df.shape[0]}, Columns: {df.shape[1]}")
df.dtypes.to_frame("dtype")

Rows: 33551, Columns: 30


,dtype
Date,str
IdUsager,int64
PV,int64
Arrondissement,int64
Mode,str
Catégorie,str
Gravité,str
Age,float64
Genre,str
Milieu,str


**Observations:** The dataset has **33,551 rows** and **30 columns**. Most columns are of type `str` (object), while `Age`, `Blessés Légers`, `Blessés hospitalisés`, and `Tué` are `float64` (nullable numerics). `Date` is stored as a string rather than `datetime64`, which will need conversion for time-series analysis. `Longitude` and `Latitude` are also stored as strings and will need conversion to float for geospatial work.

## 2. Null counts per column

In [7]:
null_counts = df.isnull().sum().to_frame("null_count")
null_counts["pct"] = (null_counts["null_count"] / len(df) * 100).round(1)
null_counts.sort_values("null_count", ascending=False)

,null_count,pct
Tué,33328,99.3
Blessés hospitalisés,31358,93.5
Adresse 2,27014,80.5
Adresse 1,23317,69.5
Coordonnées,10235,30.5
Nom arrondissement,10235,30.5
Blessés Légers,2384,7.1
Résumé,1020,3.0
Adresse,3,0.0
Date,0,0.0


**Observations:** The columns with the highest null rates are severity indicators: `Tué` is 99.3% null and `Blessés hospitalisés` is 93.5% null — this is expected since most accidents only have minor injuries. `Adresse 1` (69.5%) and `Adresse 2` (80.5%) are largely missing, suggesting address data was not consistently collected across years. `Coordonnées` and `Nom arrondissement` share the same 30.5% null rate (10,235 rows), indicating a consistent subset of records lacks this metadata. The core analytical columns (`Date`, `Mode`, `Gravité`, `Genre`, `Age`, `Milieu`, `Arrondissement`) are fully populated with 0 nulls.

## 3. Unique values for categorical columns

In [8]:
for col in ["Mode", "Catégorie", "Gravité", "Genre", "Milieu", "Tranche d'age"]:
    vals = df[col].unique()
    print(f"{col} ({len(vals)} unique): {vals}\n")

Mode (5 unique): <StringArray>
['2 Roues Motorisées', 'EDP-m', 'Piéton', 'Vélo', '4 Roues']
Length: 5, dtype: str

Catégorie (3 unique): <StringArray>
['Conducteur', 'Piéton', 'Passager']
Length: 3, dtype: str

Gravité (3 unique): <StringArray>
['Blessé léger', 'Blessé hospitalisé', 'Tué']
Length: 3, dtype: str

Genre (2 unique): <StringArray>
['Masculin', 'Feminin']
Length: 2, dtype: str

Milieu (2 unique): <StringArray>
['En-Agg', 'Hors-Agg']
Length: 2, dtype: str

Tranche d'age (9 unique): <StringArray>
[  '25-34 ans',   '18-24 ans',   '45-54 ans',   '35-44 ans',   '55-64 ans',
   '65-74 ans', '75 ans et +',    '0-13 ans',    '14-17ans']
Length: 9, dtype: str



**Observations:** The categorical columns have well-defined value sets: 5 transport modes (motorized two-wheelers, non-motorized personal mobility devices, pedestrians, bicycles, and 4-wheel vehicles), 3 user categories (driver, pedestrian, passenger), 3 severity levels, 2 genders, and 2 environments (urban/suburban). The 9 age brackets span from children (0-13) to seniors (75+). Note that `EDP-m` stands for *Engins de Déplacement Personnel motorisés* (motorized personal mobility devices like e-scooters), which is a newer category reflecting evolving urban transport.

## 4. Duplicate column analysis

The file has multiple columns that appear to be duplicates from bad joins. Let's quantify overlap.

In [9]:
# Coordonnées vs Coordonnées.1 vs Coordonnées.2
print("=== Coordonnées ===")
print(f"Coordonnées nulls:        {df['Coordonnées'].isnull().sum()}")
print(f"Coordonnées.1 nulls:       {df['Coordonnées.1'].isnull().sum()}")
print(f"Coordonnées.2 nulls:       {df['Coordonnées.2'].isnull().sum()}")
print(f"Coordonnées.1 == .2 (all): {(df['Coordonnées.1'] == df['Coordonnées.2']).all()}")
mask = df['Coordonnées'].notna()
print(f"Where non-null, .1 == .2:  {(df.loc[mask, 'Coordonnées'] == df.loc[mask, 'Coordonnées.1']).all()}")
print(f"Rows where . is null but .1 has data: {df['Coordonnées'].isnull().sum() - df['Coordonnées.1'].isnull().sum()}")

=== Coordonnées ===
Coordonnées nulls:        10235
Coordonnées.1 nulls:       0
Coordonnées.2 nulls:       0
Coordonnées.1 == .2 (all): True
Where non-null, .1 == .2:  True
Rows where . is null but .1 has data: 10235


**Observations:** `Coordonnées.1` and `Coordonnées.2` are 100% identical and fully populated, while `Coordonnées` has 10,235 nulls. Where `Coordonnées` is non-null, it matches `.1` exactly. This confirms `.1` and `.2` are redundant duplicates — `Coordonnées.1` should be used as the canonical coordinate column since it has no missing values.

In [10]:
# Nom arrondissement vs .1 vs .2
print("=== Nom arrondissement ===")
print(f"Nom arrondissement nulls:        {df['Nom arrondissement'].isnull().sum()}")
print(f"Nom arrondissement.1 nulls:       {df['Nom arrondissement.1'].isnull().sum()}")
print(f"Nom arrondissement.2 nulls:       {df['Nom arrondissement.2'].isnull().sum()}")
print(f".1 == .2 (all):                  {(df['Nom arrondissement.1'] == df['Nom arrondissement.2']).all()}")

=== Nom arrondissement ===
Nom arrondissement nulls:        10235
Nom arrondissement.1 nulls:       0
Nom arrondissement.2 nulls:       0
.1 == .2 (all):                  True


**Observations:** Same pattern as coordinates: `Nom arrondissement.1` and `.2` are identical and complete, while the original `Nom arrondissement` is missing 10,235 values. Use `Nom arrondissement.1` as the canonical column.

In [11]:
# Arrondissement vs .1 vs .2
print("=== Arrondissement ===")
print(f"Arrondissement == Arrondissement.1: {(df['Arrondissement'] == df['Arrondissement.1']).all()}")
print(f"Arrondissement == Arrondissement.2: {(df['Arrondissement'] == df['Arrondissement.2']).all()}")
print(f"Arrondissement.2 dtype: {df['Arrondissement.2'].dtype}")
print(f"Arrondissement.2 unique values: {sorted(df['Arrondissement.2'].dropna().unique())[:10]}...")
suspect = df[df['Arrondissement'].astype(str) != df['Arrondissement.2'].astype(str)]
print(f"Rows where .2 doesn't match: {len(suspect)}")

=== Arrondissement ===
Arrondissement == Arrondissement.1: True
Arrondissement == Arrondissement.2: False
Arrondissement.2 dtype: float64
Arrondissement.2 unique values: [np.float64(75.0), np.float64(75101.0), np.float64(75102.0), np.float64(75103.0), np.float64(75104.0), np.float64(75105.0), np.float64(75106.0), np.float64(75107.0), np.float64(75108.0), np.float64(75109.0)]...
Rows where .2 doesn't match: 33551


**Observations:** `Arrondissement` and `Arrondissement.1` are identical (both `int64`). However, `Arrondissement.2` is a `float64` with different values — it includes a mix of department codes (75) and full arrondissement codes (75101–75120), causing 33,551 mismatches. `Arrondissement.2` appears to come from a different join and should be dropped; use `Arrondissement` (or `.1`) as the canonical column.

In [12]:
# Adresse vs Adresse 1 vs Adresse 2
print("=== Adresse ===")
print(f"Adresse nulls:     {df['Adresse'].isnull().sum()}")
print(f"Adresse 1 nulls:   {df['Adresse 1'].isnull().sum()}")
print(f"Adresse 2 nulls:   {df['Adresse 2'].isnull().sum()}")
mask = df['Adresse'].notna() & df['Adresse 1'].notna()
print(f"Adresse == Adresse 1 (where both non-null): {(df.loc[mask, 'Adresse'] == df.loc[mask, 'Adresse 1']).sum()}/{mask.sum()}")
print(f"\nThese are DIFFERENT address components, not duplicates.")

=== Adresse ===
Adresse nulls:     3
Adresse 1 nulls:   23317
Adresse 2 nulls:   27014
Adresse == Adresse 1 (where both non-null): 4395/10232

These are DIFFERENT address components, not duplicates.


**Observations:** The three address columns are not duplicates — they represent different components. `Adresse` is the primary address (almost complete, only 3 nulls), while `Adresse 1` and `Adresse 2` are cross-street / secondary address fields with high null rates (69.5% and 80.5%). Among the 10,232 rows where both `Adresse` and `Adresse 1` are non-null, only 43% match, confirming they hold different street names (likely the two streets at an intersection).

In [13]:
# arrondgeo check
print(f"arrondgeo nulls: {df['arrondgeo'].isnull().sum()}")
print(f"arrondgeo sample type: {type(df['arrondgeo'].iloc[0])}")
print(f"arrondgeo contains GeoJSON MultiPolygon polygons for each arrondissement.")
print(f"Always populated — pure spatial metadata, not a textual duplicate.")

arrondgeo nulls: 0
arrondgeo sample type: <class 'str'>
arrondgeo contains GeoJSON MultiPolygon polygons for each arrondissement.
Always populated — pure spatial metadata, not a textual duplicate.


**Observations:** `arrondgeo` contains GeoJSON `MultiPolygon` geometries defining the boundary of each Paris arrondissement. It is always populated and is purely spatial metadata used for mapping — not a duplicate of the textual arrondissement name columns.

## 5. Date range & year distribution

In [14]:
print(f"Min date: {df['Date'].min()}")
print(f"Max date: {df['Date'].max()}\n")
df["year"] = df["Date"].str[:4].astype(int)
df["year"].value_counts().sort_index().to_frame("row_count")

Min date: 2019-01-01
Max date: 2024-12-31



,row_count
year,
2019,6357
2020,5338
2021,5836
2022,5785
2023,5429
2024,4806


**Observations:** The dataset covers **January 2019 to December 2024** (6 years). The number of person-records ranges from ~4,800 to ~6,400 per year. The peak is 2019 (6,357) with a notable dip in 2020 (5,338), likely due to COVID-19 lockdowns reducing traffic. 2024 shows the lowest count (4,806), which may indicate incomplete data for the year. Note that each row is a *person* involved in an accident, not an accident itself — multi-person accidents inflate these counts.

## 6. Injury indicator consistency check

In [15]:
# Check that each row has exactly one injury flag set
flags = df[["Blessés Légers", "Blessés hospitalisés", "Tué"]]
flag_sum = flags.notna().sum(axis=1)
print("Rows with 0 flags set:", (flag_sum == 0).sum())
print("Rows with 1 flag set:", (flag_sum == 1).sum())
print("Rows with 2+ flags set:", (flag_sum >= 2).sum())

# Verify they match Gravité
df["_bl"] = df["Blessés Légers"].notna()
df["_bh"] = df["Blessés hospitalisés"].notna()
df["_tu"] = df["Tué"].notna()
df["_match"] = (
    ((df["Gravité"] == "Blessé léger") & df["_bl"]) |
    ((df["Gravité"] == "Blessé hospitalisé") & df["_bh"]) |
    ((df["Gravité"] == "Tué") & df["_tu"])
)
print(f"\nGravité matches its indicator column: {df['_match'].all()}")
df.drop(columns=["_bl", "_bh", "_tu", "_match"], inplace=True)

Rows with 0 flags set: 0
Rows with 1 flag set: 33519
Rows with 2+ flags set: 32

Gravité matches its indicator column: True


**Observations:** Each row has at least one injury flag set, and 33,519 out of 33,551 rows (99.9%) have exactly one flag — confirming the three injury indicator columns are mutually exclusive. The 32 rows with 2+ flags are rare anomalies (likely data entry errors). The `Gravité` column is perfectly consistent with its corresponding indicator column. These three flags can serve as binary features or be used to validate the `Gravité` field.

## 7. Duplicate rows check

In [16]:
print(f"Completely duplicate rows: {df.duplicated().sum()}")
print(f"Duplicate rows excluding arrondgeo: {df.drop(columns=['arrondgeo']).duplicated().sum()}")

Completely duplicate rows: 0
Duplicate rows excluding arrondgeo: 0


**Observations:** There are zero completely duplicate rows, even when excluding the bulky `arrondgeo` column. Every record in the dataset is unique, so no deduplication step is needed.

## 8. Résumé column samples

In [17]:
print(f"Total rows: {len(df)}")
print(f"Unique non-null values: {df['Résumé'].nunique()}")
print(f"Null values: {df['Résumé'].isnull().sum()}\n")

for i in [0, 7, 14, 39]:
    print(f"--- Row {i} ---")
    print(df["Résumé"].iloc[i])
    print()

Total rows: 33551
Unique non-null values: 19937
Null values: 1020

--- Row 0 ---
Accident Léger non mortel, En agglomération, En X, Plein jour, avec une météo Normale et une surface chaussée : Normale. 

1 Véhicule de tourisme (VT) circulant sur Voie Communale VC (VMA à 30) conduit par 1 usager Masculin de 42 ans (Ind)

heurte 1 Moto

--- Row 7 ---
Accident Léger non mortel, En agglomération, Hors intersection, Plein jour, avec une météo Normale et une surface chaussée : Mouillée. 

1 Scooter  > 50 <= 125 cm3 circulant sur Voie Communale VC (VMA à 70) conduit par 1 usager Feminin de 47 ans (BL)

heu

--- Row 14 ---
Accident Léger non mortel, En agglomération, En X, Plein jour, avec une météo Normale et une surface chaussée : Normale. 

1 VU seul 1,5T < PTAC <=3,5T circulant sur Voie Communale VC (VMA à 50) conduit par 1 usager Masculin de 68 ans (Ind)

heurte 1 Scoo

--- Row 39 ---
Accident Léger non mortel, En agglomération, Place, Nuit avec éclairage public  allumé, avec une météo No

**Observations:** The `Résumé` column contains free-text accident descriptions in French. Each entry summarizes the accident context (severity, location type, weather, road surface) and describes the vehicles and users involved. With 19,937 unique values across 32,531 non-null rows, many summaries are shared across multiple persons involved in the same accident. This column is valuable for NLP tasks such as accident classification or keyword extraction, but some entries are truncated.